info/container.go

```diff
@@ -405,3 +405,36 @@ func calculateCpuUsage(prev, cur uint64) uint64 {
 	}
 	return cur - prev
 }
+
+type Percentiles struct {
+	// Indicates whether the stats are present or not.
+	Present bool `json:"present"`
+	// Average over the collected sample.
+	Mean uint64 `json:"mean"`
+	// Max seen over the collected sample.
+	Max uint64 `json:"max"`
+	// 90th percentile over the collected sample.
+	Ninety uint64 `json:"ninety"`
+}
+
+type Usage struct {
+	// Indicates amount of data available [0-100].
+	// If we have data for half a day, we'll still process DayUsage,
+	// but set PercentComplete to 50.
+	PercentComplete int32 `json:"percent_complete"`
```

Comment:
It's 50% of all stats. That's why it is part of Usage.

On Wed, Feb 4, 2015 at 9:22 AM, Vish Kannan <notifications@github.com>
wrote:

> In info/container.go
> <https://github.com/google/cadvisor/pull/481#discussion_r24100741>:
>
> > +type Percentiles struct {
> > +	// Indicates whether the stats are present or not.
> > +	Present bool `json:"present"`
> > +	// Average over the collected sample.
> > +	Mean uint64 `json:"mean"`
> > +	// Max seen over the collected sample.
> > +	Max uint64 `json:"max"`
> > +	// 90th percentile over the collected sample.
> > +	Ninety uint64 `json:"ninety"`
> > +}
> > +
> > +type Usage struct {
> > +	// Indicates amount of data available [0-100].
> > +	// If we have data for half a day, we'll still process DayUsage,
> > +	// but set PercentComplete to 50.
> > +	PercentComplete int32 `json:"percent_complete"`
>
> My concern is the latter. When we say 50% data available, is it 50% for
> all stats, which is how I interpret the current API, or 50% of one of the
> stats? To make it unambiguous, we can move the PercentComplete to
> 'Percentiles'.
> … <#14b559dac968e296_>
> On Wed, Feb 4, 2015 at 6:17 AM, Rohit Jnagal <notifications@github.com>
> wrote: In info/container.go <
> https://github.com/google/cadvisor/pull/481#discussion_r24085185>: >
> +type Percentiles struct { > + // Indicates whether the stats are present
> or not. > + Present bool `json:"present"` > + // Average over the collected
> sample. > + Mean uint64 `json:"mean"` > + // Max seen over the collected
> sample. > + Max uint64 `json:"max"` > + // 90th percentile over the
> collected sample. > + Ninety uint64 `json:"ninety"` > +} > + > +type Usage
> struct { > + // Indicates amount of data available [0-100]. > + // If we
> have data for half a day, we'll still process DayUsage, > + // but set
> PercentComplete to 50. > + PercentComplete int32 `json:"percent_complete"`
> PercentComplete just indicates how many samples cadvisor has collected for
> that time duration. The primary use for present is when the system is
> configured to say not have memory isolation or cpu load, for example. Are
> you concerned about cadvisor failing sometimes to collect a single resource
> data? We'll handle that internally in cadvisor either by ignoring the
> sample of extrapolating. It should not bubble up to the clients. — Reply to
> this email directly or view it on GitHub <
> https://github.com/google/cadvisor/pull/481/files#r24085185>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/cadvisor/pull/481/files#r24100741>.
>